src/praisonai/praisonai/integrations/db_session_adapter.py

```
@@ -137,26 +141,34 @@ def _purge_persisted_metadata(self, session_id: str) -> None:
-                    self._history_cache[session_id] = []
-                if session_id not in self._metadata:
-                    loaded = self._load_metadata_from_db(session_id)
-                    if loaded:
-                        self._metadata[session_id] = loaded
-                self._sessions.add(session_id)
-            except Exception as e:
-                logger.warning("[db_session_adapter] on_agent_start failed: %s", e)
+        # Call the DB outside the lock — it may be slow or re-enter this adapter.
+        try:
+            msgs = self._db.on_agent_start(agent_name, session_id)
+        except Exception as e:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)